crates/rpc/rpc/src/eth/cache/mod.rs

```diff
@@ -599,39 +598,15 @@ where
 
 /// All message variants sent through the channel
 enum CacheAction {
-    GetBlockWithSenders {
-        block_hash: B256,
-        response_tx: BlockWithSendersResponseSender,
-    },
-    GetBlockTransactions {
-        block_hash: B256,
-        response_tx: BlockTransactionsResponseSender,
-    },
-    GetEnv {
-        block_hash: B256,
-        response_tx: EnvResponseSender,
-    },
-    GetReceipts {
-        block_hash: B256,
-        response_tx: ReceiptsResponseSender,
-    },
-    BlockWithSendersResult {
-        block_hash: B256,
-        res: ProviderResult<Option<BlockWithSenders>>,
-    },
-    ReceiptsResult {
-        block_hash: B256,
-        res: ProviderResult<Option<Arc<Vec<Receipt>>>>,
-    },
-    EnvResult {
-        block_hash: B256,
-        res: Box<ProviderResult<(CfgEnvWithHandlerCfg, BlockEnv)>>,
-    },
-    CacheUpdateNewCanonicalChain {
-        blocks: Vec<SealedBlockWithSenders>,
-        receipts: Vec<BlockReceipts>,
-        is_new: bool,
-    },
+    GetBlockWithSenders { block_hash: B256, response_tx: BlockWithSendersResponseSender },
+    GetBlockTransactions { block_hash: B256, response_tx: BlockTransactionsResponseSender },
+    GetEnv { block_hash: B256, response_tx: EnvResponseSender },
+    GetReceipts { block_hash: B256, response_tx: ReceiptsResponseSender },
+    BlockWithSendersResult { block_hash: B256, res: ProviderResult<Option<BlockWithSenders>> },
+    ReceiptsResult { block_hash: B256, res: ProviderResult<Option<Arc<Vec<Receipt>>>> },
+    EnvResult { block_hash: B256, res: Box<ProviderResult<(CfgEnvWithHandlerCfg, BlockEnv)>> },
+    CacheNewCanonicalChain { blocks: Vec<SealedBlockWithSenders>, receipts: Vec<BlockReceipts> },
+    UncacheOldCanonicalChain { blocks: Vec<SealedBlockWithSenders>, receipts: Vec<BlockReceipts> },
```

Comment:
I want to add a new helper type for these arguments: blocks, receipts

```rust
struct ChainChange {blocks, receipts}
```

then we can simplify the duplicated code below:

```rust

fn new(Arc<Chain>)-> ChainChange 

```

which we can reuse for both reorg and committed